internal/rbd/replicationcontrollerserver.go

```diff
@@ -437,6 +529,30 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 		}
 	}
 
+	var mode librbd.ImageMirrorMode
```

Comment:
Shouldn't the whole block from 532 all the way to 540 be inside the block above? IOW, do you need to put it starting from line 529?

Reply:
The reason for keeping it out of the loop is if the csidriver is restarted after promote and before enabling the mirroring on the dummy image on the next request the if check will be skipped as the image is already primary